pkg/tsdb/prometheus/query_parser.go

```diff
@@ -0,0 +1,97 @@
+package prometheus
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/tsdb"
+)
+
+type QueryModel struct {
+	Expr           string `json:"expr"`
+	LegendFormat   string `json:"legendFormat"`
+	Interval       string `json:"interval"`
+	IntervalMS     int    `json:"intervalMS"`
+	StepMode       string `json:"stepMode"`
+	RangeQuery     bool   `json:"range"`
+	InstantQuery   bool   `json:"instant"`
+	IntervalFactor int    `json:"intervalFactor"`
+	StepMS         int    `json:"stepMS"`
+}
+
+func (s *Service) parseQuery(dsInfo *DatasourceInfo, queryContext *backend.QueryDataRequest) (
+	[]PrometheusQuery, error) {
+	qs := []PrometheusQuery{}
+	for _, query := range queryContext.Queries {
+		model := &QueryModel{}
+		err := json.Unmarshal(query.JSON, model)
+		if err != nil {
+			return nil, err
+		}
+
+		step := time.Duration(model.StepMS) * time.Millisecond
+		if step == time.Duration(0) {
```

Comment:
```suggestion
		if model.StepMS == 0 {
```
nit, but maybe that would be more clear?